### ISSUE TITLE
Support custom headers

### ISSUE THREAD

**Issue by fabasoad** [CamiloGarciaLaRotta/watermelon-http-client#4]
Would be great to support custom headers (so I can use this action in my project 😄 ).

**Comment by CamiloGarciaLaRotta** [comment]
happy to get that working! PRs are also welcome 🙌🏽

I was thinking of a key-valye format. Would this work for you @fabasoad ? 🤔

```yml
uses: CamiloGarciaLaRotta/watermelon-http-client@v1
with:
  url: <some-url>
  method: post
  data: <some-data>
  headers:
    Accept: "audio/*; q=0.2, audio/basic"
    Cache-Control: no-cache
    Foo-Header: "!*$#"
    
```

**Comment by CamiloGarciaLaRotta** [comment]
This would also address https://github.com/CamiloGarciaLaRotta/watermelon-http-client/issues/3

**Comment by CamiloGarciaLaRotta** [comment]
huh apparently hash/array inputs are currently [not supported](https://github.community/t5/GitHub-Actions/Can-action-inputs-be-arrays/m-p/33804), but we can use the following workaround:

```yml
uses: CamiloGarciaLaRotta/watermelon-http-client@v1
with:
  url: <some-url>
  method: post
  data: <some-data>
  headers: |
    {
      'Accept': ' "audio/*; q=0.2, audio/basic" ',
      'Cache-Control': 'no-cache',
      'Foo-Header': '!*$#'
    }
```

**Comment by fabasoad** [comment]
Looks good. Thank you! 🥇

**Comment by CamiloGarciaLaRotta** [comment]
@fabasoad #25 adds support for custom headers. I'll release the new version after it lands in master.
In the meantime, if you want to try it out, you can use it with:

```yml
uses: CamiloGarciaLaRotta/watermelon-http-client@feat/headers
```

Here's an [example](https://github.com/CamiloGarciaLaRotta/watermelon-http-client/pull/26/files#diff-727e6f759b67d062a68e502ae19b08bcR18) of the action using secrets and custom headers.

**Comment by CamiloGarciaLaRotta** [comment]
shipped [`v1.2`](https://github.com/CamiloGarciaLaRotta/watermelon-http-client/releases/tag/v1.2)
